src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -1000,6 +1019,35 @@ okapiHeaders, null, handlePreviousResponse(false, false, false,
     }
   }
 
+  private javax.ws.rs.core.Response loginResponseLegacy(Response loginResponse, CompositeUser cu) {
+    String token = loginResponse.getHeaders().get(OKAPI_TOKEN_HEADER);
+    return PostBlUsersLoginResponse.respond201WithApplicationJson(cu,
+      PostBlUsersLoginResponse.headersFor201().withXOkapiToken(token));
+  }
+
+  private javax.ws.rs.core.Response loginResponse(Response loginResponse, CompositeUser cu) {
+    JsonObject body = loginResponse.getBody();
+    String accessTokenExpiration = body.getString("refreshTokenExpiration");
+    String refreshTokenExpiration = body.getString("accessTokenExpiration");
+    var tokenExpiration = new TokenExpiration();
+    tokenExpiration.setAccessTokenExpiration(accessTokenExpiration);
+    tokenExpiration.setRefreshTokenExpiration(refreshTokenExpiration);
+    cu.setTokenExpiration(tokenExpiration);
+
+    List<String> setCookieHeaders = loginResponse.getHeaders().getAll("Set-Cookie");
+    var setCookieHeadersArray = setCookieHeaders.toArray();
+
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key, which is what we need
+    // here. This is a permanent workaround as long as mod-users-bl uses RMB.
+    return javax.ws.rs.core.Response.status(201)
+        .header("Set-Cookie", setCookieHeadersArray[0])
+        .header("Set-Cookie", setCookieHeadersArray[1])
```

Comment:
This fails if there is a third cookie.
Can you simply pass on all cookies?

Reply:
I very much wanted to just pass the headers on, but sadly the `org.folio.rest.tools.client.Response` uses a `MultiMap` to represent its headers, and the `javax.ws.rs.core.Response` has a `replaceAll` method that wants a `MultivaluedMap`. I'm open to suggestions if you can see an alternative.

> If this fails there is a third cookie

Not sure I'm understanding this one. Can you elaborate?

Reply:
```
    // Use the ResponseBuilder rather than RMB-generated code because
    // RMB generated-code does not allow multiple headers with the same key, which is what we need
    // here. This is a permanent workaround as long as mod-users-bl uses RMB.
    var responseBuilder = javax.ws.rs.core.Response.status(201)
        .type(MediaType.APPLICATION_JSON)
        .entity(cu);
    for (String cookie : loginResponse.getHeaders().getAll("Set-Cookie")) {
      responseBuilder.header("Set-Cookie", cookie);
    }
    return responseBuilder.build();
```

Reply:
If there are more than 2 cookies the first 2 cookies are passed on, all other cookies are dropped. Can we be sure that the dropped cookies are not needed?

Reply:
Yes agree. Made change.